src/parser/formatting-evaluator-module.ts

```diff
@@ -139,17 +141,36 @@ export class FormattingEvaluatorModule implements Module {
 
     for (const element of elements) {
       const tagName = element.tagName.toLowerCase();
-      const wordCount = this._countWords(this._multipliers[commentType].regex, element.textContent || "");
+      // We cannot use textContent otherwise we would duplicate counts, so instead we extract text nodes
+      const textNodes = Array.from(element?.childNodes || []).filter((node) => node.nodeType === 3);
+      const innerText = textNodes
+        .map((node) => node.nodeValue?.trim())
+        .join(" ")
+        .trim();
+      const symbols = this._countSymbols(this._multipliers[commentType].regex, innerText);
       let score = 0;
       if (this._multipliers[commentType]?.html[tagName] !== undefined) {
         score = this._multipliers[commentType].html[tagName];
       } else {
         logger.error(`Could not find multiplier for comment [${commentType}], <${tagName}>`);
       }
-      tagWordCount[tagName] = {
-        symbols: wordCount,
-        score,
-      };
+      logger.debug("Tag content results", { tagName, symbols, text: element.textContent });
+      // If we already had that tag included in the result, merge them and update total count
```

Comment:
I suppose that for the statistics it might be interesting to count words per element but honestly its out of scope and doesn't add business value while complicating the code.

Reply:
Will be changed in https://github.com/ubiquibot/conversation-rewards/issues/92